@here/harp-materials/lib/MapMeshMaterials.ts

```diff
@@ -407,6 +413,183 @@ export class FadingFeatureMixin implements FadingFeature {
+        );
+
+        shader.vertexShader = insertShaderInclude(
+            shader.vertexShader,
+            "begin_vertex",
+            "extrusion_vertex",
+            true
+        );
+    }
+
+    /**
+     * @param object [[THREE.Object3D]] to prepare for rendering.
+     * @param additionalCallback If defined, this function will be called before the function will
+     *          return.
+     */
+    export function addRenderHelper(object: THREE.Object3D) {
+        object.onBeforeRender = chainCallbacks(
+            object.onBeforeRender,
+            ExtrusionFeature.onBeforeRender
+        );
+    }
+
+    export function onBeforeRender(
+        renderer: THREE.WebGLRenderer,
+        scene: THREE.Scene,
+        camera: THREE.Camera,
+        geometry: THREE.Geometry | THREE.BufferGeometry,
+        material: THREE.Material,
+        group: THREE.Group
+    ) {
+        const extrusionMaterial = material as ExtrusionFeature;
+        const properties = renderer.properties.get(material);
+
+        if (
+            properties.shader !== undefined &&
+            properties.shader.uniforms.extrusionRatio !== undefined
+        ) {
+            properties.shader.uniforms.extrusionRatio.value =
+                extrusionMaterial.extrusionRatio || AnimatedExtrusionHandler.DEFAULT_RATIO_MAX;
+            extrusionMaterial.uniformsNeedUpdate = true;
+        }
+    }
+}
+
+/**
+ * Mixin class for extended THREE materials. Adds new properties required for `extrusionRatio`.
+ * Thre is some special handling for the extrusionRatio property, which get some setters and
+ * getters in a way that works well with the mixin.
+ *
+ * @see [[Tile#addRenderHelper]]
+ */
+export class ExtrusionFeatureMixin implements ExtrusionFeature {
+    needsUpdate?: boolean;
+    uniformsNeedUpdate?: boolean;
+    private m_extrusion: number = AnimatedExtrusionHandler.DEFAULT_RATIO_MAX;
+
+    /**
+     * @see [[ExtrusionFeature#extrusion]]
+     */
+    protected getExtrusionHeight(): number {
+        return this.m_extrusion;
+    }
+    /**
+     * @see [[ExtrusionFeature#extrusion]]
+     */
+    protected setExtrusionHeight(value: number) {
+        this.needsUpdate = this.needsUpdate || value !== this.m_extrusion;
+        this.m_extrusion = value;
+        if (this.needsUpdate) {
+            ExtrusionFeature.updateExtrusionFeature(this);
+        }
+    }
+
+    /**
+     * The mixin class should call this method to register the property [[extrusionRatio]]
+     */
+    protected addExtrusionProperties(): void {
+        Object.defineProperty(this, "extrusionRatio", {
+            get: () => {
+                return this.getExtrusionHeight();
+            },
+            set: val => {
+                this.setExtrusionHeight(val);
+            }
+        });
+    }
+
+    /**
+     *
+     * Apply the extrusionRatio value from the parameters to the respective properties.
+     *
+     * @param params `ExtrusionMeshBasicMaterial` parameters.
```

Comment:
if you there is no way to write meaningful  param doc and signature says it all, just don't document it ...